File: cpp/src/Hole.cpp
#include <Hole.h>

int Hole::harvest(){
    int currentPopulation = population;
    population = 0;
    return currentPopulation;
}

int Hole::seed(int numSeeds){
    population += 1;
    return numSeeds - 1;
}

Hole::Hole(int population): population(population), rumba(false)
{
}

Hole::Hole(int population, bool rumba):
    population(population), rumba(rumba)
{
}

File: julia/hole.jl
abstract type BaseHole end

struct Hole <: BaseHole
    diameter::Real
    population::Int

    function Hole(population::Int)
        new(1,population)
    end
end

struct Rumba <: BaseHole
    diameter::Real
    population::Int

    function Rumba(population::Int)
        new(4,population)
    end
end

function Base.show(io::IO, hole::Hole)
    print(io, " $(hole.population) ")
end

function Base.show(io::IO, hole::Rumba)
    print(io, " $(hole.population) ")
end